src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/PackagesLockFileUtilities.cs

```diff
@@ -172,9 +172,18 @@ public static bool IsLockFileStillValid(DependencyGraphSpec dgSpec, PackagesLock
                             // The package spec not found in the dg spec. This could mean that the project does not exist anymore.
                             if (p2pSpec != null)
                             {
-                                // This does not consider ATF.
-                                var p2pSpecTargetFrameworkInformation = NuGetFrameworkUtility.GetNearest(p2pSpec.TargetFrameworks, framework.FrameworkName, e => e.FrameworkName);
-
+                                TargetFrameworkInformation p2pSpecTargetFrameworkInformation = default;
+                                if (p2pSpec.RestoreMetadata.ProjectStyle == ProjectStyle.PackagesConfig || p2pSpec.RestoreMetadata.ProjectStyle == ProjectStyle.Unknown)
+                                {
+                                    // Skip compat check and dependency check for non PR projects.
+                                    // Projects that are not PR do not undergo compat checks by NuGet and do not contribute anything transitively.
+                                    p2pSpecTargetFrameworkInformation = p2pSpec.TargetFrameworks.FirstOrDefault();
```

Comment:
Can you achieve multitargeting with a packages.config project? If that's the case, would it be necessary to perform a framework compatibility check in this case?

Reply:
No, packages.config is only started with classic csproj and that on it's own *does not* support multi targeting. 
Multi targeting was first added in project.json and now SDK based projects are the *only* actively supported project type that supports multi targeting.